docs/tutorials/cloud-init.md

```diff
@@ -0,0 +1,192 @@
+the latest LTS application as well.
+
+## What you will learn:
+
+- How to write cloud-config user data to a specific WSL instance.
+- How to automatically setup a WSL instance with cloud-init.
+- How to verify that cloud-init succeeded with the configuration supplied.
+
+## What you will need:
+
+- Windows 11 with WSL 2 already enabled
+- The latest UbuntuPreview application from Microsoft Store.
+
+## Write the cloud-config file
+
+Locate your Windows user home directory. It typically is `C:\Users\<YOUR_USER_NAME>`.
+
+> You can be sure about that path by running `echo $env:USERPROFILE` in PowerShell.
+
+Inside your Windows user home directory, create a new folder named `.cloud-init` (notice the `.` à la Linux
+configuration directories), and inside it create a new empty file named `Ubuntu-Preview.user-data`. That file name must
+match the name of the distro instance that will create in the next step.
+
+Open that file with your text editor of choice (`notepad.exe` is just fine) and paste in the following contents:
+
+```yaml
+#cloud-config
+locale: pt_BR
+users:
+- name: u
+  gecos: Ubuntu
+  groups: [adm,dialout,cdrom,floppy,sudo,audio,dip,video,plugdev,netdev]
+  sudo: ALL=(ALL) NOPASSWD:ALL
+  shell: /bin/bash
+
+write_files:
+- path: /etc/wsl.conf
+  append: true
+  content: |
+    [user]
+    default=u
+
+packages: [ginac-tools, octave]
+
+runcmd:
+   - git clone https://github.com/Microsoft/vcpkg.git /opt/vcpkg
+   - /opt/vcpkg/bootstrap-vcpkg.sh
+```
+
+Save it and close it.
+
+> That example will create a user named `u` and set it as default via `/etc/wsl.conf`, install the packages
+> `ginac-tools` and `octave` and install `vcpkg` from the git repository, since there is no deb or snap of that
+> application (hence the reason for being included in this tutorial - it requires an unusual setup).
+
+## Register a new Ubuntu-Preview instance
+
+In PowerShell, run:
+
+```powershell
+ubuntupreview.exe install --root
+```
+We skip the user creation, since we expect cloud-init to do it.
+
+> If you want to be sure that there is now an Ubuntu-Preview instance, run `wsl -l -v`.
+> Notice that the application is named `UbuntuPreview` but the WSL instance created is named `Ubuntu-Preview`.
+
+## Check that cloud-init is running
+
+In PowerShell again run:
+
+
+```powershell
+ubuntupreview run cloud-init status --wait
+```
+
+That will wait until cloud-init completes configuring the new instance we just created. When done, you should see an
+output similar to the following:
+
+```powershell
+..............................................................................
+..............................................................................
+..............................................................................
+...................
+status: done
+```
+
+
+## Verify that it worked
+
+Restart the distro just to make sure the changes in `/etc/wsl.conf` made by cloud-init will take effect, as listed
```

Comment:
More feature-related than documentation-related: Cloud we have cloud-init drop the "please restart" file for the launcher? Installs will be done with a launcher most of the time.

Reply:
I don't think we should because then we'd have cloud-init assuming the ubuntu launcher, when cloud-init is meant to work with multiple distros.